pkg/replication/replication.go

```diff
@@ -266,6 +272,10 @@ func (r replication) handleRenewalErrors(session kp.Session, renewalErrCh chan e
 	select {
 	case <-r.replicationDoneCh:
 	case <-r.replicationCancelledCh:
+		// If the replication is enacted, wait for it to exit
+		select {
+		case <-r.enactedCh:
```

Comment:
this is equivalent to `<-r.enactedCh`. I think you probably meant to select on `r.quitCh` in here as well